audio and voice effects doc for windows (#959)
AgoraIO/Docs-Source: shared/video-sdk/develop/audio-and-voice-effects/project-implementation/windows.mdx

    The resource opens inside the **Dialog Editor**. 

3. To open **Toolbox**, go to menu **View** > **ToolBox**.

4. From **ToolBox**, drag **Button** to the surface of the **Dialog Editor**.

    You see a button in your **Dialog Editor**.

6. Repeat the step four and add two more buttons to the UI.

### Handle the system logic

This section describes the steps required to set up access to the UI elements.

1. **Define variables to manage audio effects and access the UI elements**

    In `AgoraImplementationDlg.h`, add the following declarations to the `CAgoraImplementation` class:

    ```cpp
	int soundEffectId = 1; // Unique identifier for the sound effect file
	std::string soundEffectFilePath = "https://www.soundjay.com/human/applause-01.mp3"; // URL or path to the sound effect
	int soundEffectStatus = 0;
	int voiceEffectIndex = 0;
	boolean audioPlaying = false; // Manage the audio mixing state
	std::string audioFilePath = "https://www.kozco.com/tech/organfinale.mp3"; // URL or path to the audio mixing file
	CButton* playEffectButton, *voiceEffectButton, *audioMixingButton;
	CButton* speakerphoneSwitch;
    ```

1. **Set up access to the speakerphone switch**

>     In `AgoraImplementationDlg.cpp`, add the following lines to the `OnInitDialog` method after `setupVideoSDKEngine();`

Reviewer: ```suggestion
    In `AgoraImplementationDlg.cpp`, add the following lines to the `OnInitDialog` method after `setupVideoSDKEngine();`:
```

    <ProductWrapper product = "interactive-live-streaming">
    ```cpp
    audioMixingButton = (CButton*)GetDlgItem(IDC_BUTTON3);
    audioMixingButton->SetWindowText(L"Mix Audio");
	playEffectButton = (CButton*)GetDlgItem(IDC_BUTTON4);
    playEffectButton->SetWindowText(L"Audio Effect");
	voiceEffectButton = (CButton*)GetDlgItem(IDC_BUTTON5);
    voiceEffectButton->SetWindowText(L"Voice Effect");
    ```
    </ProductWrapper>
    <ProductWrapper product = "video-calling">
    ```cpp
    speakerphoneSwitch = (CButton*)GetDlgItem(IDC_CHECK1);
    audioMixingButton = (CButton*)GetDlgItem(IDC_BUTTON3);
    audioMixingButton->SetWindowText(L"Mix Audio");
	playEffectButton = (CButton*)GetDlgItem(IDC_BUTTON4);    
    playEffectButton->SetWindowText(L"Audio Effect");
	voiceEffectButton = (CButton*)GetDlgItem(IDC_BUTTON5);
    voiceEffectButton->SetWindowText(L"Voice Effect");
    ```
    </ProductWrapper>

### Implement sound and voice effects

To add sound and voice effect to your <Vpl k="CLIENT" />, take the following steps:

1. **Enable the user to start and stop audio mixing**
        

>     When the user presses **Mix Audio**, the <Vg k = "VSDK"/> fetches an audio file using `audioFilePath` and mixes it with the local audio stream. The remote and local users listen the audio file playing with the audio stream. To enable audio mixing in your <Vpl k = "CLIENT"/>, in **Dialog Editor**, double-click **Mix Audio**. **Dialog Editor** automatically creates and opens an event listener for you. Add the following code to the event listener you just created:

Reviewer: ```suggestion
    When the user clicks **Mix Audio**, the <Vg k = "VSDK"/> fetches an audio file using `audioFilePath` and mixes it with the local audio stream. The remote and local users listen the audio file playing with the audio stream. To enable audio mixing in your <Vpl k = "CLIENT"/>, in **Dialog Editor**, double-click **Mix Audio**. **Dialog Editor** automatically creates and opens an event listener for you. Add the following code to the event listener you just created:
```

      
    ```cpp
    audioPlaying = ~audioPlaying;
	if (audioPlaying) 
    {
		audioMixingButton->SetWindowText(L"Stop Mixing");
		agoraEngine->startAudioMixing(audioFilePath.c_str(), false, 1, 0);
	}
	else
	{
		agoraEngine->stopAudioMixing();
		audioMixingButton->SetWindowText(L"Mix Audio");
	}
    ```

1. **Pre-load sound effects**

> 

Reviewer: Google style guide advises to avoid the hyphen and use a closed form of the verb, unless it is one of the exceptions: https://developers.google.com/style/hyphens

    To set up playing voice effects, call `preloadEffect` to pre-load the sound effects. In `setupVideoSDKEngine`, add the following lines after `agoraEngine->initialize(context);`:

    ```cpp
    // Pre-load sound effects to improve performance
	agoraEngine->preloadEffect(soundEffectId, soundEffectFilePath.c_str());
    ```

1. **Play, pause, or resume playing the sound effect**

>     When a user presses **Audio Effect**, the sound effect is played. When they press the button again, the effect is paused and resumed alternately. To implement this workflow, in **Dialog Editor**, double-click **Audio Effect**. **Dialog Editor** automatically creates and opens an event listener for you. Add the following code to the event listener you just created:

Reviewer: ```suggestion
    When a user clicks **Audio Effect**, the sound effect is played. When they click the button again, the effect is paused and resumed alternately. To implement this workflow, in **Dialog Editor**, double-click **Audio Effect**. **Dialog Editor** automatically creates and opens an event listener for you. Add the following code to the event listener you just created:
```

	{ 
        // Stopped
        agoraEngine->playEffect(
			soundEffectId,   // The ID of the sound effect file.
			soundEffectFilePath.c_str(),   // The path of the sound effect file.
			0,  // The number of sound effect loops. -1 means an infinite loop. 0 means once.
			1,   // The pitch of the audio effect. 1 represents the original pitch.
			0.0, // The spatial position of the audio effect. 0.0 represents that the audio effect plays in the front.
			100, // The volume of the audio effect. 100 represents the original volume.
			true,// Whether to publish the audio effect to remote users.
			0    // The playback starting position of the audio effect file in ms.
		);
		playEffectButton->SetWindowText(L"Pause effect");
		soundEffectStatus = 1;
	}
	else if (soundEffectStatus == 1) 
    { // Playing
		agoraEngine->pauseEffect(soundEffectId);
		soundEffectStatus = 2;
		playEffectButton->SetWindowText(L"Resume effect");
	}
	else if (soundEffectStatus == 2) 
    { // Paused
		agoraEngine->resumeEffect(soundEffectId);
		soundEffectStatus = 1;
		playEffectButton->SetWindowText(L"Pause effect");
    }
    ```

1. **Update the UI after the effect finishes playing**

>     When <Vpd k="SDK" /> has finished playing the sound effect, the `onAudioEffectFinished` event is fired. Your <Vpl k = "CLIENT"/> listens to the callback and invoke the `OnEIDStopAudioEffect` method to update the UI and stop the audio effect. To Implement this logic, take the following steps:

Reviewer: ```suggestion
    When <Vpd k="SDK" /> has finished playing the sound effect, the `onAudioEffectFinished` event is fired. Your <Vpl k = "CLIENT"/> listens to the callback and invokes the `OnEIDStopAudioEffect` method to update the UI and stop the audio effect. To implement this logic, take the following steps:
```

        ```

    1. In `AgoraImplementationDlg.cpp`, add the following before `OnInitDialog`:

        ```cpp
        LRESULT CAgoraImplementation::OnEIDStopAudioEffect(WPARAM soundId, LPARAM lParam)
        {
            AfxMessageBox(L"Audio effect finished");
            agoraEngine->stopEffect(soundId);
            soundEffectStatus = 0; // Stopped
            playEffectButton->SetWindowText(L"Audio effect");
            return 0;
        }
        ```

    1. In `AgoraImplementationDlg.cpp`, add the following before `ON_WM_CLOSE()`:

        ```cpp
        ON_MESSAGE(WM_MSGID(EID_STOP_EFFECT), &CAgoraImplementation::OnEIDStopAudioEffect)
        ```

1. **Set an audio profile**

    For applications where special audio performance is required, you set a suitable audio profile and audio scenario. In `setupVideoSDKEngine`, add the following lines after `agoraEngine->initialize(context);`:

        ```cpp
        // Specify the audio scenario and audio profile
        agoraEngine->setAudioProfile(AUDIO_PROFILE_MUSIC_HIGH_QUALITY_STEREO);
	    agoraEngine->setAudioScenario(AUDIO_SCENARIO_GAME_STREAMING);
        ```

1. **Apply various voice and audio effects**

>     When a user presses **Voice Effect**, apply a new voice effect and change the text on the button to describe the effect. To implement this workflow, in **Dialog Editor**, double-click **Voice Effect**. **Dialog Editor** automatically creates and opens an event listener for you. Add the following code to the event listener you just created:

Reviewer: ```suggestion
    When a user clicks **Voice Effect**, apply a new voice effect and change the text on the button to describe the effect. To implement this workflow, in **Dialog Editor**, double-click **Voice Effect**. **Dialog Editor** automatically creates and opens an event listener for you. Add the following code to the event listener you just created:
```
